tests/dummy/app/routes/application.js

```diff
@@ -2,7 +2,16 @@ import Ember from 'ember';
 // app/routes/application.js
 export default Ember.Route.extend({
   intl: Ember.inject.service(),
-  beforeModel () {
-    return this.get('intl').setLocale('en-us');
+  application: Ember.inject.service(),
+
+  beforeModel (/* transition */) {
+    return this.get('application').authenticate()
+      .then(authData => {
+        const state = authData.state;
+        if (state && state !== '/') {
+          this.transitionTo(state);
+        }
```

Comment:
This isn't the smoothest transition when coming back from auth. I think it does end up trying to `transitionTo` the same route twice. I didn't want to spend a ton of time working on coming back from auth and redirecting to the right page. This was the quickest, easiest way to get it done without just dropping them back to the home route (which is also fine for such a small app).